### ISSUE TITLE
[Feature Request]: BrowserView.getFromWebContents

### ISSUE THREAD

**Issue by Willyfrog** [electron/electron#27686]
**Preflight Checklist**

* [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for a feature request that matches the one I want to file, without success.

**Problem Description**

some events get a webcontents parameter and this one might be from a browserview. trying to the window with `BrowserWindow.getFromWebcontents(webcontents)` returns null when it belongs to a BrowserView.
Looking into `webcontents.getType()` will tell you that it belongs to the browserview but there seems to be no known way to get the browserview out of it. For that I had to iterate all windows, and on each iterate each browserview attached to figure out which one is the one that caused the event.

You can see some sample code in [this electron-dl's PR](https://github.com/sindresorhus/electron-dl/pull/128)

**Proposed Solution**

Being able to do
```js
session.on('will-download', (event, item, webcontents) => {
  const currentBrowserView = BrowserView.getFromWebContents(webcontents);
  const currentWindow = BrowserWindow.getFromBrowserView(currentBrowserView);
});
```

**Alternatives Considered**

have the option to get the window out of the webcontents received, but I felt it would be less useful and only especific for te particular case I was trying to fix.

**Additional Information**

_No response_

**Comment by nornagon** [comment]
> BrowserWindow.getFromWebcontents(webcontents) returns null when it belongs to a BrowserView

this was fixed here: https://github.com/electron/electron/pull/25408

**Comment by Willyfrog** [comment]
> > BrowserWindow.getFromWebcontents(webcontents) returns null when it belongs to a BrowserView
> 
> this was fixed here: #25408

hey! thanks for pointing it out, it looks like it would solve the problem. But I can't find in the releases page any mention to it, which version does that change land? is it only in v11?

**Comment by nornagon** [comment]
@Willyfrog the earliest version that PR is included in is v12.0.0-beta.1.

**Comment by Willyfrog** [comment]
thank you very much :)
